### PR TITLE
adding modestimage recipe

### PR DESCRIPTION
**Pull request by kbeeperez**
implements: https://github.com/nsls-ii-forge/modestimage-feedstock

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kbeeperez** [comment]
Attn @mrakitin 

I removed this from the source section because it failed the first time around with an error showing that it couldn't find it. Let me know if this needs to be put back in.
```
  patches:
    - mpl31.patch
```

**Review by mrakitin — CHANGES_REQUESTED** [review]
The patch is still needed, as the project is not actively maintained anymore, and no future contributions may be accepted. See https://github.com/ChrisBeaumont/mpl-modest-image/issues/10 and https://github.com/ChrisBeaumont/mpl-modest-image/pull/12 for details.

To get the patch, you'll need to download the raw version of https://github.com/nsls-ii-forge/modestimage-feedstock/blob/master/recipe/mpl31.patch, and revert back the reference to the patch in `meta.yaml`.

**Comment by mrakitin** [comment]
To @conda-forge/staged-recipes, I agree to be a maintainer of this feedstock.

File: recipes/modestimage/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - matplotlib
```
Comment (dopplershift) [inline recipes/modestimage/meta.yaml:24]: Looks like this doesn't need Qt:
```suggestion
    - matplotlib-base
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
This is just about ready to go in, just looks like `matplotlib` could be replaced with `matplotlib-base`.

File: recipes/modestimage/meta.yaml (hunk 1 above)
Comment (mrakitin) [reply recipes/modestimage/meta.yaml:24]: I don't think we need Qt for that package, good idea.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**